addons/sale_margin/tests/test_sale_margin.py

```diff
@@ -151,3 +152,33 @@ def test_sale_margin3(self):
         # Verify that margin field gets bind with the value.
         self.assertEqual(sale_order_so14.margin, 100.00, "Sales order profit should be 100.00")
         self.assertEqual(sale_order_so14.margin_percent, 0.4, "Sales order margin should be 40%")
+
+    def test_sale_margin_order_copy(self):
+        """When we copy a sales order, its margins should be update to meet the current costs"""
+        # We buy at a specific price today and our margins go according to that
+        self.product.standard_price = 500.0
+        original_sale = self.SaleOrder.create({
+            'partner_id': self.partner_id,
+            'partner_invoice_id': self.partner_invoice_address_id,
+            'partner_shipping_id': self.partner_invoice_address_id,
+            'pricelist_id': self.pricelist_id,
+            'order_line': [
+                Command.create({
+                    'price_unit': 1000.0,
+                    'product_uom': self.product_uom_id,
+                    'product_uom_qty': 10.0,
+                    'product_id': self.product.id,
+                }),
+            ],
+        })
+        original_sale.action_confirm()
```

Comment:
No need to confirm the sales order. Save test time.

Reply:
Removed :+1: